metricbeat/module/kibana/kibana_integration_test.go

```diff
@@ -0,0 +1,65 @@
+// Licensed to Elasticsearch B.V. under one or more contributor
+// license agreements. See the NOTICE file distributed with
+// this work for additional information regarding copyright
+// ownership. Elasticsearch B.V. licenses this file to you under
+// the Apache License, Version 2.0 (the "License"); you may
+// not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+// +build integration
+
+package kibana_test
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/require"
+
+	"github.com/elastic/beats/libbeat/tests/compose"
+	mbtest "github.com/elastic/beats/metricbeat/mb/testing"
+	"github.com/elastic/beats/metricbeat/module/kibana"
+	_ "github.com/elastic/beats/metricbeat/module/kibana/stats"
+)
+
+var xpackMetricSets = []string{
+	"stats",
+}
+
+func TestXPackEnabled(t *testing.T) {
+	service := compose.EnsureUpWithTimeout(t, 300, "kibana")
+
+	metricSetToTypeMap := map[string]string{
+		"stats": "kibana_stats",
+	}
+
+	config := getXPackConfig(service.Host())
+
+	metricSets := mbtest.NewReportingMetricSetV2Errors(t, config)
```

Comment:
I think we prefer using [`NewFetcher`](https://github.com/elastic/beats/blob/8c71abc3dd960e510ae34d91e464fcd913676455/metricbeat/mb/testing/fetcher.go#L40) like [here](https://github.com/elastic/beats/blob/4def200842e0cf7540343d14276b103be7fa20fe/metricbeat/module/kafka/broker/broker_integration_test.go#L52)

Reply:
Good to know about this, thanks. But it looks like `NewFetcher` will return a single metricset whereas over here I need to fetch multiple metricsets based on one config that specifies them all. 

This is because the stack modules are a bit special: when the `xpack.enabled: true` configuration option is set in a stack module's configuration, there is some [special validation](https://github.com/elastic/beats/blob/4f89dcd0ffc56f42865973ba466730946e0c78a1/metricbeat/module/kibana/kibana.go#L44) that kicks in to ensure that certain metricsets are specified in the configuration. Now, in the case of the `kibana` module, there is only [one metricset](https://github.com/elastic/beats/blob/master/metricbeat/module/kibana/kibana.go#L67) that is required when `xpack.enabled: true` is set so I could use `NewFetcher` over here. But this is not generally the case for other stack modules (e.g. [`logstash`](https://github.com/elastic/beats/blob/master/metricbeat/module/logstash/logstash.go#L64-L67), [`elasticsearch`](https://github.com/elastic/beats/blob/master/metricbeat/module/elasticsearch/elasticsearch.go#L68-L78), and [`beat`](https://github.com/elastic/beats/blob/master/metricbeat/module/beat/beat.go#L63-L66)). And I would prefer that all stack modules' integration tests follow the same pattern. So I would like to leave this line of code as-is.

Reply:
👍